build.bash

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env bash
+
+nproc_default=4
+
+script_name=$(basename "${0}")
+
+show_help() {
+    cat << EOF
+Usage: ./$script_name [OPTIONS]
+
+Build script wrapper around CMake. Supplied arguments that do not match the
+options below will be passed to CMake when generating the build system.
+
+Options:
+      --clean   Delete build directory before invoking CMake.
+      --mpi     Compile MPI executable.
+      --compiler <compiler>
+                Specify the compiler to use.
+  -j <jobs>     Specify the number of parallel jobs in the compilation. By
+                default this value is set to $nproc_default.
+  -h, --help    Show this screen.
+
+Enabling debug mode:
+
+  The release build is default. To enable debug mode, specify the CMake option
+  -DCMAKE_BUILD_TYPE=Debug when invoking $script_name.
+
+Enabling verbose output from Makefile builds:
+
+  To enable more verbose output from Makefile builds, specify the CMake option
+  -DCMAKE_VERBOSE_MAKEFILE=ON when invoking $script_name.
+
+EOF
+}
+
+cmake_args=(-DCMAKE_BUILD_TYPE=Release)
+
+# Argument parsing adapted and stolen from http://mywiki.wooledge.org/BashFAQ/035#Complex_nonstandard_add-on_utilities
+while [ $# -gt 0 ]; do
+    case $1 in
+        --clean)
+            rm -r build
+            ;;
+        --mpi)
+            mpi=1
+            cmake_args+=(-DCABLE_MPI="ON")
+            ;;
+        --compiler)
+            compiler=$2
+            shift
+            ;;
+        -j)
+            CMAKE_BUILD_PARALLEL_LEVEL=$2
+            shift
```

Comment:
I changed my opinion ;-)  I think that `-j` is only talkative if you know make. But who knows make? It is now also the only option that has a single dash. Perhaps `--ncpus`? But see my comment about single dash-double dash options.

Reply:
I'd be happy to go with `--ncpus` as it is more descriptive.